CommonItem/src/main/java/com/github/sef24sp4/common/item/loottable/LootTable.java

```diff
@@ -0,0 +1,103 @@
+package com.github.sef24sp4.common.item.loottable;
+
+import com.github.sef24sp4.common.item.CommonItem;
+import com.github.sef24sp4.common.item.ItemRarity;
+import com.github.sef24sp4.common.item.ItemSPI;
+
+import java.util.*;
+
+public class LootTable {
+	private final Map<ItemRarity, Double> itemChances;
+	private final Map<ItemRarity, Collection<ItemSPI>> providers;
+
+	/**
+	 * Generates a map from {@link ItemRarity} to a collection of {@link ItemSPI} instances.
+	 *
+	 * @param providers A collection of item providers.
+	 * @return A map from {@link ItemRarity} to a collection of {@link ItemSPI} instances.
+	 */
+	private static Map<ItemRarity, Collection<ItemSPI>> generateItemProvidersMap(final Collection<ItemSPI> providers) {
+		final Map<ItemRarity, Collection<ItemSPI>> map = new HashMap<>();
```

Comment:
Change from `Map<ItemRarity, Collection<ItemSPI>>` to type `Map<ItemRarity, Liste<ItemSPI>>`